sky/spot/controller.py

```diff
@@ -271,6 +276,35 @@ def _run_one_task(self, task_id: int, task: 'sky.Task') -> bool:
                                      task_id,
                                      recovered_time=recovered_time)
 
+    def event_callback_func(self, task_id: int, state: str, comment: str = ''):
+
+        callback_str = self._dag.tasks[task_id].event_callback
+        if callback_str is None:
+            return
+
+        if '$JOB_ID' in callback_str:
+            callback_str = callback_str.replace('$JOB_ID', str(self._job_id))
+        if '$TASK_ID' in callback_str:
+            callback_str = callback_str.replace('$TASK_ID', str(task_id))
+        if '$JOB_STATUS' in callback_str:
+            callback_str = callback_str.replace('$JOB_STATUS', state)
+        if '$COMMENT' in callback_str:
+            callback_str = callback_str.replace('$COMMENT', comment)
+
+        if callback_str.find('controller_log:') == 0:
+            print_str = callback_str[len('controller_log:'):].strip()
+            logger.info(print_str)
```

Comment:
Maybe we can get rid of this, as the controller log is normally not a user-facing log, and we probably have enough loggings in it.
If a user would like to log something to the controller log, they can also specify the bash script with `echo hello world` in it.